zsh/configs/post/path.zsh

```diff
@@ -1,9 +1,17 @@
 # ensure dotfiles bin directory is loaded first
 PATH="$HOME/.bin:/usr/local/sbin:$PATH"
 
-# load ASDF, falling back to rbenv if not available
-if [ -d "$HOME/.asdf" ]; then
-  . $HOME/.asdf/asdf.sh
+# Try loading ASDF from the regular home dir location
+if [ -f "$HOME/.asdf/asdf.sh" ]; then
+  . "$HOME/.asdf/asdf.sh"
+# It's not in the home dir, let's try the default Homebrew location
+elif [ -f "/usr/local/opt/asdf/asdf.sh" ]; then
+  . "/usr/local/opt/asdf/asdf.sh"
+# Not there either! Last try, let's see if Homebrew can tell us where it is
+elif which brew >/dev/null &&
+  ASDF_PREFIX=$(brew --prefix asdf) &&
```

Comment:
This works but `brew` takes about a second to return here. I'd love to figure out a faster way to check whether a package is installed.

Reply:
That's pretty slow. Is that `brew` startup time? Or is it searching something? Is `brew ls --versions asdf` faster?

Is it looping through all your installed packages? https://github.com/Homebrew/brew/blob/14755541b0e786e190d415a6d9d74c2e5023802a/Library/Homebrew/extend/ARGV.rb#L29-L35

What if we made a claim like:

> If you changed your Homebrew installation path away from the default, you now have to deal with the consequences: a slower shell startup time. Set `ASDF_PREFIX` to improve this again.

And then start the entire conditional off with trying to source `"$ASDF_PREFIX/asdf.sh"`.

?

Reply:
Sadly all `brew` commands take around a second to come back, and it doesn't set any environment variables that we can check either.

I wonder if it might be better to leave this part out entirely? Then if somebody installs Homebrew somewhere else and installs `asdf` it's up to them to source the completions file in their local config file.